### PR TITLE
feat: integrates storybook

### LINKED ISSUE #121: Storybook integration
## Expected Behavior
Documentation and visual testing for components

## Actual Behavior
No docs, no visual test

## Steps to Reproduce the Problem
N/A

## Specifications
N/A

### PR DESCRIPTION
**Pull request by guilhermevrs**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?
No visual documentation for the documents

Issue Number: Closes #121 

## What is the new behavior?
This PR integrates storybook with examples on the BaseCard component

A couple of changes:

- In order to correct have addons-actions to work, no default value for callback parameters
- Run `yarn storybook` to kick the local storybook server
- Fixed minor issues with the eslint
- No coverage for stories

## Does this introduce a breaking change?

- [X] Yes
- [ ] No

As described above, in order to have addons-actions to work, callback parameters should not have default values

## Other information

In order to restain the size of this PR, I have migrated only BaseCard with examples. We would need to open another story to migrate the other ones.
Also, we would need as well to integrate a way of publishing the storybook result (like with Chromatic)

## A picture tells a thousand words

image

### CODE REVIEW

File: packages/components/src/components/atoms/BaseCard/BaseCard.tsx [hunk 1]
```diff
@@ -95,8 +95,8 @@ export const BaseCard: FunctionComponent<BaseCardProps> = ({
   tableBoundaries,
   disableNativeEvents,
   position = { x: 0, y: 0 },
-  onPositionChanged = () => {},
-  onFlipped = () => {},
+  onPositionChanged,
+  onFlipped,
```
Comment (tib-tib) [inline packages/components/src/components/atoms/BaseCard/BaseCard.tsx:99]: Why did you have to change this? If you need to change it then maybe we should also set the property as not optional 😄

File: packages/components/src/components/atoms/BaseCard/stories/Pure.story.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+import { defaultStoryTemplate, defaultStoryMetaFactory } from './common';
+
+export const PureExample = defaultStoryTemplate.bind({});
+PureExample.args = {
+  frontFace: <div>Front face</div>,
+  backFace: <div>Back face</div>,
+};
+
+export default defaultStoryMetaFactory({ title: 'Pure' });
```
Comment (tib-tib) [inline packages/components/src/components/atoms/BaseCard/stories/Pure.story.tsx:11]: I don't think we need this file, or do we? What is it doing exactly

File: packages/components/src/components/organisms/Table/index.tsx [hunk 3]
```diff
@@ -1 +1,2 @@
+/* istanbul ignore file */
```
Comment (tib-tib) [inline packages/components/src/components/organisms/Table/index.tsx:1]: Why do we need this? We should rather configure jest to exclude `index.tsx` files in my opinion

File: packages/components/tsconfig.json [hunk 4]
```diff
@@ -10,6 +10,7 @@
     "moduleResolution": "node",
     "allowSyntheticDefaultImports": true,
     "esModuleInterop": true,
+    "strictBindCallApply": true,
```
Comment (tib-tib) [inline packages/components/tsconfig.json:13]: Can you explain me why you needed to add this option? 😄

File: packages/components/src/components/atoms/BaseCard/BaseCard.tsx (hunk 1 above)
Comment (guilhermevrs) [reply packages/components/src/components/atoms/BaseCard/BaseCard.tsx:99]: Yeah, with the default value for callbacks, Storybook actions (the panel that logs on event activation) seems to not work - it's on the PR description

The events should still to be optional IMO. Why exactly do you want to make them not optional?

File: packages/components/src/components/atoms/BaseCard/stories/Pure.story.tsx (hunk 2 above)
Comment (guilhermevrs) [reply packages/components/src/components/atoms/BaseCard/stories/Pure.story.tsx:11]: It's a story showing we can use `BaseCard` with whatever you want as face (so if you want to use another type of card).
Doesn't make sense?

I advise you to run locally to see the stories running to get a better feeling of what is doing

File: packages/components/src/components/organisms/Table/index.tsx (hunk 3 above)
Comment (guilhermevrs) [reply packages/components/src/components/organisms/Table/index.tsx:1]: Yeah - I was wondering if we want to remove all the `index.tsx` files from coverage or not...
`table` slices for example have all the logic on the `index.tsx`.

Shall we move it then and remove all the index.tsx from the coverage?

File: packages/components/tsconfig.json (hunk 4 above)
Comment (guilhermevrs) [reply packages/components/tsconfig.json:13]: Yes, it's to ensure that `.bind({})` for functions enforce the typings.

Example here: https://github.com/drunkoders/cards/pull/125/files#diff-5a75b3b5d5cff2d3c58ea81dd1b63d9c86952e87c9575cd67352c3dfa1472659R5

File: packages/components/src/components/organisms/Table/index.tsx (hunk 3 above)
Comment (tib-tib) [reply packages/components/src/components/organisms/Table/index.tsx:1]: Yeah I would suggest to move it and remove all index from coverage. We can do it in another PR 😄

**Pull request closed without merging**